plugins/modules/inventory_workflow_manager.py

```diff
@@ -2562,12 +2561,14 @@ def update_interface_detail_of_device(self, device_to_update):
         """
 
         # Call the Get interface details by device IP API and fetch the interface Id
+        is_update_occurred = False
         for device_ip in device_to_update:
             interface_params = self.config[0].get('update_interface_details')
             interface_names_list = interface_params.get('interface_name')
             for interface_name in interface_names_list:
                 device_id = self.get_device_ids([device_ip])
-                interface_id = self.get_interface_from_id_and_name(device_id[0], interface_name)
+                interface_details = self.get_interface_from_id_and_name(device_id[0], interface_name)
+                interface_id = interface_details['id']
```

Comment:
Do this only if get_interface_from_id_and_name returns..

```
interface_details = self.get_interface_from_id_and_name(device_id[0], interface_name)

# Check if interface_details is None or does not contain the 'id' key.
if interface_details is None or 'id' not in interface_details:
    # Handle the case where no details are found or 'id' is missing.
    self.log("Unable to obtain interface details or 'id' not present for device '{0}' with interface '{1}'.".format(device_id[0], interface_name))
   set self and retun from here.............
    return self

interface_id = interface_details['id']
```

Reply:
added the check and mark status as failed.